extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -1129,7 +1129,12 @@ private void setupUnixDomainSocketHttpServer(HttpServer httpServer, HttpServerOp
                         startFuture.complete(null);
                     }
                 } else {
-                    startFuture.fail(event.cause());
+                    if (event.cause() instanceof IllegalArgumentException) {
+                        startFuture.fail(new IllegalArgumentException(
+                                "Unable to bind to Unix domain socket. Consider adding the 'io.netty:netty-transport-native-epoll' dependency. See the Quarkus Vert.x reference guide for more details."));
```

Comment:
There is also `netty-transport-native-kqueue`.

What about extending the message a bit:

```suggestion
                                "Unable to bind to Unix domain socket. Consider adding the 'io.netty:netty-transport-native-epoll' or 'io.netty:netty-transport-native-kqueue' dependency. See the Quarkus Vert.x reference guide for more details."));
```